### DISCUSSION TITLE
Horizon Armor System

### DISCUSSION THREAD

**Discussion by MasterDrake** [jrouwe/JoltPhysics#1035] [General]
Hi, I'm planning to make a game around the same armor and weak points system in the Horizon games and I was wondering if you could give me some pointers on how to and how it was done.
Especially how the physical representation of the armor/piece is linked to the mesh and skeleton, considering the animations move everything around.
Are the physics objects mesh shapes, so it's easier to make sure the physical representation is easily derived from the mesh and it's easier somehow to reason in vertices and skin bones/joints/matrixes?
How are hierarchical weak points made? The ones that (partially or not) cover other weak points.
What are the challenges of ensuring that these pieces are positioned at their place and don't get away, especially because of rounding errors and floating point precision?

Thanks and sorry to bother you!

**Comment by jrouwe** [comment]
I'm not sure in how much detail I can go about this, but it's not very complex. Each part has an associated Body that is usually a Kinematic ConvexHull and moves along with the animation. Parts that are behind other parts probably ignore damage until their covering part is destroyed / only exist visually until that time (I didn't set up this logic).

**Comment by MasterDrake** [comment]
> I'm not sure in how much detail I can go about this, but it's not very complex. Each part has an associated Body that is usually a Kinematic ConvexHull and moves along with the animation. Parts that are behind other parts probably ignore damage until their covering part is destroyed / only exist visually until that time (I didn't set up this logic).
Thanks for the answer :D
So you're saying that with jolt it's not hard to create the physic bodies and then make them follow an animation?
What's the closest example I can dig in?

**Reply by jrouwe** [reply]
That would be https://github.com/jrouwe/JoltPhysics/blob/master/Samples/Tests/General/KinematicTest.cpp

Edit: Instead of calling SetLinearVelocity, there's also BodyInterface::MoveKinematic where you can specify the target position and the velocity will be calculated for you.

**Comment by MasterDrake** [comment]
> That would be https://github.com/jrouwe/JoltPhysics/blob/master/Samples/Tests/General/KinematicTest.cpp
> 
> Edit: Instead of calling SetLinearVelocity, there's also BodyInterface::MoveKinematic where you can specify the target position and the velocity will be calculated for you.
I'll get to it, thank you so much :D
